Add django backend

Blu3spirits opened the pull request:
May end up being a little overkill in the end, but I noticed a lot of the CSV entries resulting from the script are duplicated

Blu3spirits committed:
Add test Django backend

- Modify requirements to reflect django addition

Blu3spirits committed:
Remove local_settings entry in manage.py

Blu3spirits committed:
Comply with PEP8

- Remove migrations file*
- Modify serializers and viewsets
- Remove unused tests

Kiro47 commented on course_viewer/course_viewer/models.py:
```diff
@@ -0,0 +1,44 @@
+                                    help_text='',
+                                    unique=True)
+    college_name = models.CharField(max_length=200,
+                                    unique=True)
```
College names aren't explicitly unique, only the codes are.  There are some colleges with the exact same name.

Blu3spirits committed:
Change how settings are handled

- Add migrations again per https://docs.djangoproject.com/en/1.7/topics/migrations/
- Add settings directory
- Add local/dev/prod settings files

Kiro47 commented on course_viewer/course_viewer/models.py:
```diff
@@ -0,0 +1,44 @@
+    '''
+    course_name = models.CharField(max_length=200)
+    course_credit = models.IntegerField()
+    course_number = models.IntegerField()
```
Course numbers aren't always number integers.  For example in our current CSV data on row 77 we have values such as `102HC`  as "course_number".  Perhaps course_id would be more appropriate?  Will probably change that in the script on my next PR.

Kiro47 commented on course_viewer/course_viewer/models.py:
```diff
@@ -0,0 +1,44 @@
+    '''
+    The MTU Course
+    '''
+    mtu_course_name = models.IntegerField()
```
Course name should probably be a `CharField`.

Blu3spirits committed:
Fix mtu-course.id not found bug

- Remove migrations dir tracking since it should be generated upon
deployment

Blu3spirits committed:
Add in final hooks for ForeignKey relations

Blu3spirits committed:
Add str overrides and rename fields

Blu3spirits committed:
Add Meta class to models

Blu3spirits committed:
Serialize foreign keys to values

This will allow the output of a curl/GET request to be displayed like
the following:
```
{
    "id": 1,
    "course_mtu_course_id": "2700",
    "course_mtu_course_subject": "HU",
    "course_college_code": "10",
    "course_college_name": "FakeCollege",
    "course_state_name": "Michigan",
    "course_state_code": "MI",
    "course_name": "Psychology 101",
    "course_id": "PSY101",
    "course_credit": 3,
    "course_number": "101"
}
```
Rather than previously where it displayed the primary key of the foreign
table.

Kiro47 commented on course_viewer/course_viewer/models.py:
```diff
@@ -0,0 +1,111 @@
+                            'State',
+                            related_name='course_state_code',
+                            on_delete=models.CASCADE,
+                            help_text='The state the course is located in')
```
Basically all the variables in this could use some changing on the naming of them.  Without the help text it becomes rather unreadable and confusing without already knowing the output contents.

Kiro47 commented on course_viewer/course_viewer/serializers.py:
```diff
@@ -0,0 +1,53 @@
+                                read_only=True)
+    course_state_code = serializers.CharField(
+                                source='course_state_code.state_code',
+                                read_only=True)
```
Basically all the variables in this could use some changing on the naming of them. Without the help text (such as when they get parsed to JSON output on a curl). Become rather unreadable and confusing without already knowing the output.

{
    "id": 1,
    "course_mtu_course_id": "2700",  // Okay it's MTU's course ID
    "course_mtu_course_subject": "HU", // Okay it's MTU's subject name
    "course_college_code": "10", // Course college code? for which course the MTU or transfer?
    "course_college_name": "FakeCollege", // Same as above
    "course_state_name": "Michigan", // Same as above
    "course_state_code": "MI", // Same as above
    "course_name": "Psychology 101", // From which? MTU or the transfering? also same as above
    "course_id": "PSY101", // same as above
    "course_credit": 3, // same as above
    "course_number": "101"// same as above, also now we're calling it a course number instead of ID
}

Blu3spirits committed:
Adjust variable names and displayed information

Kiro47 commented on course_viewer/course_viewer/models.py:
```diff
@@ -0,0 +1,111 @@
+                        max_length=15,
+                        help_text='The id of the transfer course')
+    transfer_course_credit = models.IntegerField(
+                    help_text='The number of credits the transfer course is')
```
Some of the entries don't have values for `transfer_course_credit`, they just are equivalent to a class (annoyingly enough without boosting credits taken).  Will Django reflect this properly with a `None` type, or will this cause an issue?

Kiro47 replied:
Looking farther into this actually, it's the same half credit point issue as the MTU credit side annoyingly enough. `International Institute` has several half credit classes.

Blu3spirits replied:
I think 
`blank=True`
and using a Float field would be best here in that case

Kiro47 commented on course_viewer/course_viewer/models.py:
```diff
@@ -0,0 +1,111 @@
+                        max_length=50,
+                        help_text='The subject of the MTU course')
+    mtu_credits = models.IntegerField(
+                        help_text='The amount of credits the MTU course is')
```
Some of the entries for `mtu_credits` earned are half credits.  Possibly use floats, then round on display for the integers?

Blu3spirits replied:
Same as last comment, blank=True and float field

Blu3spirits replied:
I think rounding would actually harm more than be helpful here. Seeing as how a 3 credit course at X college transferring as a .5 credit but displaying as 1 credit would be misleading in planning things like graduation credit fulfillment.

Blu3spirits committed:
Change credits to allow floats

Some credits can be half credits and also transfer in as half credits

Blu3spirits committed:
Add migrations folder to git ignore

Blu3spirits committed:
Restructure Django project

- Make the project a little less redundant with the directories
- Slight rename in the project since the directory name did not match

Blu3spirits committed:
Ignore F401, F403 PEP8 errors

Ignoring these errors because in order for the settings to be structured
like this the import * is absolutely needed.
